modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/EditSegmentsEntryDisplayContext.java

```diff
@@ -348,6 +350,41 @@ private JSONObject _getInitialQueryJSONObject(
 		FilterParser filterParser = _filterParserProvider.provide(
 			segmentsCriteriaContributor.getEntityModel());
 
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-166954"))) {
+			try {
+				Expression expression = filterParser.parse(
+					criterionFilterString);
+
+				JSONObject jsonObject = (JSONObject)expression.accept(
+					new ExpressionVisitorImpl(
+						1, segmentsCriteriaContributor.getEntityModel()));
+
+				if (Validator.isNull(jsonObject.getString("groupId"))) {
+					jsonObject = JSONUtil.put(
+						"conjunctionName",
+						StringUtil.toLowerCase(
+							String.valueOf(BinaryExpression.Operation.AND))
+					).put(
+						"groupId", "group_0"
+					).put(
+						"items", JSONUtil.putAll(jsonObject)
+					);
+				}
+
+				return jsonObject;
+			}
+			catch (Exception exception) {
+				if (_log.isDebugEnabled()) {
+					_log.error(exception);
+				}
+			}
+
+			return JSONFactoryUtil.createJSONObject(
+				HashMapBuilder.put(
```

Comment:
Why the JSON returned is created using a createJSONObject, instead  JSONUtil.put directly?

Reply:
You right, JSONUtil.put works just fine. Thank u